src/PayU/PaymentsApi/PaymentsV4/Services/HashService.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+
+namespace PayU\PaymentsApi\PaymentsV4\Services;
+
+use PayU\Alu\MerchantConfig;
+use PayU\PaymentsApi\PaymentsV4\PaymentsV4;
+
+class HashService
+{
+    /**
+     * @param MerchantConfig $merchantConfig
+     * @param string $orderDate
+     * @param string $jsonRequest
+     * @return string
+     */
+    public function generateSignature($merchantConfig, $orderDate, $jsonRequest)
+    {
+        $stringToBeHashed =
+            $merchantConfig->getMerchantCode() .
+            $orderDate .
+            HTTPClient::POST_METHOD .
```

Comment:
This signature seems to be build specifically for the authorization call. In this case, it would be better to make it clear by naming its purpose (rename to generateAuthorizationSignature) or make it more generic (HTTPClient::POST_METHOD and PaymentsV4::PAYMENTS_API_AUTHORIZE_PATH sent as paramters)

Reply:
done, made it more generic as complete and refund flows will be added in the future